src/targets/furnace.js

```diff
@@ -0,0 +1,98 @@
+import { ClickableObject } from "../clickable-object.js";
+import { CONSTANTS } from "../constants/constants.js";
+import { characterData } from "../cookie-io.js";
+import { Button } from "../ui/button.js";
+import { getItemClass } from "../utilities.js";
+
+export class Furnace extends ClickableObject {
+    name = "Furnace";
+    varName = "bar"; // Use # bars forged as quest
+    examineText = "A red hot furnace.";
+    actions = [
+        { text: "Forge", func: "clickTarget" },
+        { text: "Examine", func: "examine" },
+    ];
+
+    constructor(scene) {
+        super();
+        this.scene = scene;
+
+        const cameraWidth = scene.cameras.main.width;
+        const cameraHeight = scene.cameras.main.height;
+        const x = cameraWidth / 2 - 230;
+        const y = cameraHeight / 2 - 150;
+        const width = 230;
+        const height = 150;
+
+        // Add invisible button for furnace
+        this.sprite = new Button(scene, x, y, width, height);
+        this.sprite.on("pointerdown", (pointer) => {
+            if (pointer.rightButtonDown()) {
+                this.createRightClickMenu(pointer.x, pointer.y, this.actions);
+            } else {
+                this.clickTarget();
+            }
+        });
+    }
+
+    async clickTarget() {
+        const inv = this.scene.dashboard.inventory;
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        // See if ore is selected
+        const selectedIndex = inv.curSelectedItemIndex;
+        if (selectedIndex < 0) {
+            chat.writeText("Select an ore in your inventory first");
+            return;
+        }
+
+        const selectedItem = inv.inventory[selectedIndex];
+        switch (selectedItem.name) {
+            case "Copper Ore":
+            case "Tin Ore":
+                this.smelt(
+                    ["CopperOre", "TinOre"],
+                    "BronzeBar",
+                    1,
+                    "You need at least 1 copper and 1 tin in your inventory."
```

Comment:
Maybe mention that a bronze bar requires this. "You need at least one copper and one tin ore to smelt a bronze bar." Also not sure what we've been doing so far, but I generally like spelling out numbers if I know they are less than ten.

Reply:
Now that I think about it, the error message and the required ores should also be specified in the class. I think you should be able to reduce the smelt method to only one parameter and get everything else from the item class.

Reply:
Done https://github.com/EthanVieira/rs-clicker/pull/60/commits/f58a102e21838136430a3019f6ce252ac0ec0b07